lib/provider.coffee

```diff
@@ -168,7 +170,42 @@ module.exports =
 
     attributePattern.exec(line)?[1]
 
-  getAttributeValues: (attribute) ->
+  getLocalStylesheets: (editor) ->
+    source = editor.getText()
+    while match = stylePattern.exec(source)
+      match[1]
+
+  getStylesheetsAtPath: (dir) ->
+    stylesheets = []
+    for file in fs.readdirSync(dir)
```

Comment:
This is more like proof of concept. Look on my to-do list about caching.

On Wed, Dec 30, 2015, 23:50 Steel Brain <notifications@github.com> wrote:

> In lib/provider.coffee
> <https://github.com/atom/autocomplete-html/pull/21#discussion_r48611158>:
>
> > @@ -168,7 +170,42 @@ module.exports =
> >
> >      attributePattern.exec(line)?[1]
> >
> > -  getAttributeValues: (attribute) ->
> > +  getLocalStylesheets: (editor) ->
> > +    source = editor.getText()
> > +    while match = stylePattern.exec(source)
> > +      match[1]
> > +
> > +  getStylesheetsAtPath: (dir) ->
> > +    stylesheets = []
> > +    for file in fs.readdirSync(dir)
>
> Scanning a whole project on each autocomplete call is too expensive IMO
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/autocomplete-html/pull/21/files#r48611158>.
>